src/interp.cc

```diff
@@ -912,6 +912,21 @@ ValueTypeRep<T> Add(ValueTypeRep<T> lhs_rep, ValueTypeRep<T> rhs_rep) {
   return ToRep(FromRep<T>(lhs_rep) + FromRep<T>(rhs_rep));
 }
 
+template <typename T, typename R, R rmax, R rmin>
+ValueTypeRep<T> AddSaturate(ValueTypeRep<T> lhs_rep, ValueTypeRep<T> rhs_rep) {
+  T max = static_cast<T>(rmax);
```

Comment:
Are these `static_cast`s necessary? It seems like they'll implicitly cast since `T` is always a larger size type than `R`.

Reply:
Yes, you are right. 

I just want to make the signed and unsigned number promotion in code more explicitly.

Thanks!